util/stopper.go

```diff
@@ -88,40 +92,83 @@ func (s *Stopper) AddCloser(c Closer) {
 	s.closers = append(s.closers, c)
 }
 
-// StartTask adds one to the count of tasks left to drain in the
-// system. Any worker which is a "first mover" when starting tasks
-// must call this method before starting work on a new task and must
-// subsequently invoke FinishTask() when the task is complete.
-// First movers include goroutines launched to do periodic work and
-// the kv/db.go gateway which accepts external client
-// requests.
+// A PendingTask is returned by StartTask. For simplicity this is just a
```

Comment:
that's the thing, it may not be. You need to call `Ok()` first. There's a TODO about the interface now.